scripts/collect_steps.py

```diff
@@ -0,0 +1,165 @@
+import os
+import re
+import subprocess
+
+# This script collects all the steps that are executed in the oneshot_ipa with
+# all possible configurations.
+
+IPA_ENV = [["RUST_LOG", "ipa=DEBUG"]]
+ARGS = [
+    "cargo",
+    "bench",
+    "--bench",
+    "oneshot_ipa",
+    "--no-default-features",
+    "--features=enable-benches debug-trace step-trace",
+    "--",
+    "--num-multi-bits",
+    "1",
+]
+QUERY_SIZE = 10
+PER_USER_CAP = [1, 3]
+ATTRIBUTION_WINDOW = [0, 86400]
+BREAKDOWN_KEYS = [1, 33]
+SECURITY_MODEL = ["malicious", "semi-honest"]
+ROOT_STEP_PREFIX = "protocol/alloc::string::String::run-0"
+EXPECTED_HEADER_LINES_COUNT = 4
+
+
+def set_env():
+    env = os.environ.copy()
+    for k, v in IPA_ENV:
+        env[k] = v
+    return env
+
+
+def remove_root_step_name_from_line(l):
+    return l.split(",")[0][len(ROOT_STEP_PREFIX) + 1 :]
+
+
+def to_int_or(s):
+    try:
+        return int(s)
+    except ValueError:
+        return s
+
+
+# Split string into list of strings and ints.
+#
+# E.g. "protocol/run-0/mc0/xor1" -> ["protocol/run-", 0, "/mc", 0, "/xor", 1]
+#
+# This is used to sort the steps in the natural order (e.g. 1, 2, 10, 11, 20).
+def natural_sort_key(s):
+    return [to_int_or(v) for v in re.split("(\d+)", s)]
+
+
+def collect_steps(args):
+    output = []
+
+    proc = subprocess.Popen(
+        args=args,
+        env=set_env(),
+        stdout=subprocess.PIPE,
+        bufsize=1,
+        universal_newlines=True,
+    )
+
+    count = 0
+    header = 0
+    while True:
+        line = proc.stdout.readline()
+        if not line or line == "":
+            break
+
+        if not line.startswith(ROOT_STEP_PREFIX):
+            header += 1
+            continue
+
+        output.append(remove_root_step_name_from_line(line))
+        count += 1
+
+    # safeguard against empty output
+    if count == 0:
+        print("No steps in the output")
+        exit(1)
+
+    # oneshot_ipa produces 4 header lines
```

Comment:
that is likely not going to be true very soon. 

```bash
Using random seed: 4631627385142799437 for 10000 records
Preparation complete in 3.515583ms
SemiHonest IPA for 10000 records took 72.599756667s
```

is this for bench output? We should probably make sure to add an argument to the bench to silence its output, otherwise we'll be making a bunch of efforts to keep this script up to date. 

Happy to iterate on that